devicesetup/src/main/java/io/particle/android/sdk/devicesetup/SetupCompleteIntentBuilder.java

```diff
@@ -0,0 +1,9 @@
+package io.particle.android.sdk.devicesetup;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.Nullable;
+
+public interface SetupCompleteIntentBuilder {
+    Intent buildIntent(Context ctx, @Nullable SetupResult result);
```

Comment:
Didn't you just give a perfect illustration of exactly why we can't rely on most developers recognizing the potential for a leak?  :)   I would bet that overwhelmingly, most developers, even sharp ones, won't notice the potential leak here, even with good documentation.  The problem is just too easy to stumble into, especially with no default tooling to point out the problem.

That said, the implementation awkwardness is a real issue.

What about an abstract class implementation which already provides all the Parcelable garbage, and just leaves them to implement that single method?